src/lib/AnnotationControls.ts

```diff
@@ -84,21 +84,53 @@ export default class AnnotationControls {
     private handleFullscreenExit = (): void => this.handleFullscreenChange(false);
 
     /**
-     * Region comment button click handler
+     * Deactive current control button
      */
-    private handleRegionClick = (onRegionClick: RegionHandler) => (event: MouseEvent): void => {
+    public deactivateCurrentControl = (): void => {
+        if (!this.currentActiveControl) {
+            return;
+        }
+
+        switch (this.currentActiveControl) {
+            case ANNOTATION_MODE.region:
+                this.currentActiveControl = null;
+                this.updateRegionButton();
+                break;
+            default:
+                this.currentActiveControl = null;
+        }
+    };
+
+    /**
+     * Update region button UI
+     */
+    private updateRegionButton = (): void => {
         const regionButtonElement = this.controlsElement.querySelector(`.${CLASS_REGION_BUTTON}`);
 
         if (regionButtonElement) {
-            this.isRegionActive = !this.isRegionActive;
-            if (this.isRegionActive) {
+            if (this.currentActiveControl === ANNOTATION_MODE.region) {
                 regionButtonElement.classList.add(CLASS_BUTTON_ACTIVE);
             } else {
                 regionButtonElement.classList.remove(CLASS_BUTTON_ACTIVE);
             }
         }
+    };
+
+    /**
+     * Region comment button click handler
+     */
+    private handleRegionClick = (onRegionClick: RegionHandler) => (event: MouseEvent): void => {
+        if (this.currentActiveControl === ANNOTATION_MODE.region) {
+            this.deactivateCurrentControl();
+        } else {
+            if (this.currentActiveControl) {
+                this.deactivateCurrentControl();
```

Comment:
It looks like this happens no matter what? Can we simplify and move it to the top of the method, outside of the conditions?